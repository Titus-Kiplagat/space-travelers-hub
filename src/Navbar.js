function Navbar() {
  return (
    <nav>
      Navigation
    </nav>
  );
}

export default Navbar;
