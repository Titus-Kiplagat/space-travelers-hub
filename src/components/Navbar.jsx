import { Nav, Navbar, Container } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import planet from '../assets/planet.png';

function Header() {
  return (
    <Navbar sticky="top" bg="light" expand="lg" className="border-bottom container">
      <Container fluid>
        <NavLink to="/" className="navbar-brand d-flex align-center">
          <img src={planet} alt="space travelers' hub logo" width="40" height="40" />
          <span className="ms-3 fw-bold fs-3">Space Travelers&apos; Hub</span>
        </NavLink>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Item>
              <NavLink className="nav-link active-link text-primary bold" to="/rockets">
                Rockets
              </NavLink>
            </Nav.Item>
            <Nav.Item>
              <NavLink className="nav-link active-link text-primary bold" to="/missions">
                Missions
              </NavLink>
            </Nav.Item>
            <div className="vr bold" />
            <Nav.Item>
              <NavLink className="nav-link active-link text-primary bold" to="/profile">
                My Profile
              </NavLink>
            </Nav.Item>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default Header;
