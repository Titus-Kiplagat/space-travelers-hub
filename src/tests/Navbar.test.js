import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
