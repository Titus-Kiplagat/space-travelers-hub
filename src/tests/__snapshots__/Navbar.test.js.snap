// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="border-bottom px-5 navbar navbar-expand-lg navbar-light bg-light sticky-top"
>
  <div
    className="container-fluid"
  >
    <a
      aria-current="page"
      className="navbar-brand d-flex align-center active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="space travelers' hub logo"
        height="40"
        src="planet.png"
        width="40"
      />
      <span
        className="ms-3 fw-bold fs-3"
      >
        Space Travelers' Hub
      </span>
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <a
            className="nav-link active-link text-primary bold"
            href="/rockets"
            onClick={[Function]}
          >
            Rockets
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link active-link text-primary bold"
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </div>
        <div
          className="vr bold"
        />
        <div
          className="nav-item"
        >
          <a
            className="nav-link active-link text-primary bold"
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
